tests/test_drsclient.py

```diff
@@ -0,0 +1,281 @@
+    rec2 = res2.json()
+    assert rec2["id"] == bundle_id
+    assert rec2["name"] == rec1["name"]
+
+
+def test_get_all(index_client, drs_client):
+
+    # create 100 files and bundles
+    for _ in range(100):
+        rec = create_index_record(index_client)
+        drs_client.create(bundles=[rec["did"]])
+
+    res1 = drs_client.get_all(limit=300)
+    assert res1.status_code == 200
+    rec1 = res1.json()
+    assert len(rec1["drs_objects"]) == 200
+
+    res2 = drs_client.get_all(form="bundle", limit=300)
+    assert res2.status_code == 200
+    rec2 = res2.json()
+    assert len(rec2["drs_objects"]) == 100
+
+    res3 = drs_client.get_all(form="object", limit=300)
+    assert res3.status_code == 200
+    rec3 = res3.json()
+    assert len(rec3["drs_objects"]) == 100
+
+
+def test_async_get_all(index_client, drs_client):
+
+    # create 100 files and bundles
+    for _ in range(100):
+        rec = create_index_record(index_client)
+        drs_client.create(bundles=[rec["did"]])
+
+    res1 = asyncio.run(drs_client.async_get_all(limit=300))
+    assert res1.status_code == 200
+    rec1 = res1.json()
+    assert len(rec1["drs_objects"]) == 200
+
+
+def test_layered_bundle(index_client, drs_client):
+    """
+    Bundle1
+        +-Object1
+        +-Bundel2
+            +-Object2
+            +-Object3
+    """
+    obj1 = create_index_record(index_client)
+    obj2 = create_index_record(index_client)
+    obj3 = create_index_record(index_client)
+    bnd2 = drs_client.create(bundles=[obj2["did"], obj3["did"]])
+    assert bnd2.status_code == 200
+    rec2 = bnd2.json()
+    bnd1 = drs_client.create(bundles=[obj1["did"], rec2["bundle_id"]])
+    assert bnd1.status_code == 200
+    rec1 = bnd1.json()
+    res = drs_client.get(rec1["bundle_id"], expand=True)
+    rec = res.json()
+    assert len(rec["contents"]) == 2
+    assert len(rec["contents"][1]["contents"]) == 2
+
+
+def test_delete_bundle(index_client, drs_client):
+    rec = create_index_record(index_client)
+    did = rec["did"]
+    res1 = drs_client.create(bundles=[did])
+    rec1 = res1.json()
+    bundle_id = rec1["bundle_id"]
+    assert res1.status_code == 200
+    res2 = drs_client.delete(bundle_id)
+    assert res2.status_code == 200
+    res3 = drs_client.get(bundle_id)
+    assert res3.status_code == 404
+
+
+def test_async_delete_bundle(index_client, drs_client):
+    rec = create_index_record(index_client)
+    did = rec["did"]
+    res1 = drs_client.create(bundles=[did])
+    rec1 = res1.json()
+    bundle_id = rec1["bundle_id"]
+    assert res1.status_code == 200
+    res2 = asyncio.run(drs_client.async_delete(bundle_id))
+    assert res2.status_code == 200
+    res3 = drs_client.get(bundle_id)
+    assert res3.status_code == 404
+
+
+def test_get_bundle_id_not_exist(index_client, drs_client):
+    res = drs_client.get("09ujh1f0938hj08hjf082jf08j")
+    print(res)
+    assert res.status_code == 404
+
+
+def test_no_id(index_client, drs_client):
```

Comment:
a comment explaining what is being tested would be helpful
but again, i feel like this might not be a drsclient test